topostats/tracing/dnatracing.py

```diff
@@ -81,644 +114,491 @@ def __init__(
-                    splined_trace = np.column_stack((x_new, y_new))
-
-                    try:
-                        np.add(spline_running_total, splined_trace)
-                    except NameError:
-                        spline_running_total = np.array(splined_trace)
-
-                spline_average = spline_running_total
-                self.splined_traces[dna_num] = spline_average
-                """
-
-                # can't get splining of linear molecules to work yet
-                self.splined_traces[dna_num] = self.fitted_traces[dna_num]
+                try:
+                    spline_running_total = np.add(spline_running_total, splined_trace)
+                except NameError:
+                    spline_running_total = np.array(splined_trace)
+
+            # if not self.splining_success:
+            #     continue
+
+            spline_average = np.divide(spline_running_total, [step_size, step_size])
+            del spline_running_total
+            self.splined_trace = spline_average
+            # else:
+            #    x = self.fitted_trace[:,0]
+            #    y = self.fitted_trace[:,1]
+
+            #    try:
+            #        tck, u = interp.splprep([x, y], s=0, per = 2, quiet = 1, k = 3)
+            #        out = interp.splev(np.linspace(0,1,nbr*step_size), tck)
+            #        splined_trace = np.column_stack((out[0], out[1]))
+            #        self.splined_trace = splined_trace
+            #    except ValueError: #if the trace is really messed up just delete it
+            #        self.mol_is_circular.pop(dna_num)
+            #        self.disordered_trace.pop(dna_num)
+            #        self.grain.pop(dna_num)
+            #        self.ordered_trace.pop(dna_num)
+
+        else:
+            # can't get splining of linear molecules to work yet
+            self.splined_trace = self.fitted_trace
 
     def show_traces(self):
         plt.pcolormesh(self.gauss_image, vmax=-3e-9, vmin=3e-9)
         plt.colorbar()
-        for dna_num in sorted(self.disordered_trace.keys()):
-            plt.plot(self.ordered_traces[dna_num][:, 0], self.ordered_traces[dna_num][:, 1], markersize=1)
-            plt.plot(self.fitted_traces[dna_num][:, 0], self.fitted_traces[dna_num][:, 1], markersize=1)
-            plt.plot(self.splined_traces[dna_num][:, 0], self.splined_traces[dna_num][:, 1], markersize=1)
-            # print(len(self.skeletons[dna_num]), len(self.disordered_trace[dna_num]))
-            # plt.plot(self.skeletons[dna_num][:,0], self.skeletons[dna_num][:,1], 'o', markersize = 0.8)
+        plt.plot(self.ordered_trace[:, 0], self.ordered_trace[:, 1], markersize=1)
+        plt.plot(self.fitted_trace[:, 0], self.fitted_trace[:, 1], markersize=1)
+        plt.plot(self.splined_trace[:, 0], self.splined_trace[:, 1], markersize=1)
 
         plt.show()
         plt.close()
 
     def saveTraceFigures(
```

Comment:
I'd used it a few times whilst doing this work to check what images are like, as we've discussed plotting the traces should be an additional method to the `plottingfuncs.Image` class with the ability to overlay on different plots, whether that is the original image of filtered heights, cropped images, masks (full or cropped) or just the trace on its own.

We'll have to decide how to return and amalgamate the different objects and find out what people want saving (#595), e.g. should we include unordered traces, I suspect not but it may be desirable to have them available perhaps.